web/client/components/share/SharePanel.jsx

```diff
@@ -348,19 +377,88 @@ class SharePanel extends React.Component {
                             onChange={({target})=>{
                                 const zoom = inRange(parseInt(target.value, 10), 1, 36) ? target.value : 1;
                                 this.setState({...this.state, zoom});
+                                this.updateMapView('zoom', zoom);
                             }}/>
                     </FormGroup>
-                    <Checkbox
-                        checked={this.props.settings && this.props.settings.markerEnabled}
-                        onChange={() => {
-                            this.props.onUpdateSettings({
-                                ...this.props.settings,
-                                markerEnabled: !this.props.settings.markerEnabled
-                            });
-                        }
-                        }>
-                        <Message msgId="share.marker" />
-                    </Checkbox>
+                    {
+                        this.props.mapType && this.props.mapType === 'cesium' && (
+                            <React.Fragment>
+                                <FormGroup>
+                                    <ControlLabel><Message msgId="share.heading" /></ControlLabel>
+                                    <OverlayTrigger placement="top" overlay={<Tooltip id="share-heading"><Message msgId="share.headingToolTip"/></Tooltip>}>
+                                        <Glyphicon style={{marginLeft: 5}} glyph="info-sign" />
+                                    </OverlayTrigger>
+                                    <FormControl
+                                        type="number"
+                                        name={"heading"}
+                                        min={0}
+                                        max={360}
+                                        value={
+                                            (this.state?.heading && this.setValueBoundaries(convertRadianToDegrees(this.state.heading), 0, 360)) ||
+                                            (this.props?.viewerOptions?.orientation?.heading && this.setValueBoundaries(convertRadianToDegrees(this.props.viewerOptions.orientation.heading), 0, 360))
```

Comment:
The invalid value is still saved to state as the `updateMapView` is triggered. This allows user to grab the shareUrl with invalid value set to orientation fields
**Ex**: `heading` with _450_ is converted and updated in shareUrl as _7.853981633974483_
Please restrict the value from `onChange` rather than converting in the `value` prop